### PR TITLE
Add org.kde.kaffeine

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://invent.kde.org/multimedia/kaffeine/-/merge_requests/6
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* https://invent.kde.org/multimedia/kaffeine/-/merge_requests/6 https://invent.kde.org/multimedia/kaffeine/-/merge_requests/7 https://invent.kde.org/multimedia/kaffeine/-/merge_requests/8

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build org.kde.kaffeine

**Comment by hfiguiere** [comment]
cc @flathub/kde

**Comment by Mailaender** [comment]
bot, build org.kde.kaffeine

**Comment by Mailaender** [comment]
bot, build org.kde.kaffeine

**Comment by travier** [comment]
Manifest seems good to me. Note that you can also add x-data-checker config for dependencies too, not just the application source. But we can also add it after the merge.

**Comment by barthalion** [comment]
It doesn't seem to play videos from local filesystem for me:

```
23-05-22 08:46:45.732 [Warning ] qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
[00007f90a80013b0] gl vout display error: parent window not available
[00007f90a80013b0] xcb vout display error: window not available
[00007f90a80013b0] xcb vout display error: window not available
[00007f90a80013b0] fb vout display error: cannot get terminal mode (Inappropriate ioctl for device)
[00007f90b0007610] main video output error: video output creation failed
[00007f90ba557370] main decoder error: failed to create video output
[libopenh264 @ 0x7f90ba557b40] get_buffer() failed
[libopenh264 @ 0x7f90ba557b40] Unable to allocate buffer
[00007f90ba557370] avcodec decoder error: avcodec_send_packet critical error
```

Can someone else check as well?

File: org.kde.kaffeine.json
Comment (travier) [inline org.kde.kaffeine.json:15]: Missing `--device=dri`

**Comment by travier** [comment]
Plays the sound but not the image for videos for me. Maybe because of the missing `dri` perm.

We also have a lot of:
```
23-05-22 08:58:55.888 [Warning ] kf.solid.backends.udisks2: /run/build-runtime/solid/src/solid/devices/backends/udisks2/udisksmanager.cpp#160: void Solid::Backends::UDisks2::Manager::introspect(const QString&, bool): Failed enumerating UDisks2 objects: "org.freedesktop.DBus.Error.Disconnected" 
 "Not connected to D-Bus server"
```

Maybe need some DBus access here.

File: org.kde.kaffeine.json
Comment (Mailaender) [reply org.kde.kaffeine.json]: Added

**Comment by Mailaender** [comment]
Video playback worked fine for me even without `dri`. Maybe this doesn't play well with Wayland.

**Comment by nedrichards** [comment]
Yup, there's an array of dbus calls getting filtered due to no allowlist if you run with --log-session-bus (mpris etc.)

**Comment by Mailaender** [comment]
I added some, but probably need help with this.

**Comment by travier** [comment]
bot, build org.kde.kaffeine

**Comment by travier** [comment]
Still can not play videos for me:

```
[libopenh264 @ 0x7fdd6f664a40] get_buffer() failed
[libopenh264 @ 0x7fdd6f664a40] Unable to allocate buffer
```

**Comment by tinywrkb** [comment]
> [libopenh264 @ 0x7fdd6f664a40] get_buffer() failed

This app is linked against FFmpeg? Then it can use the full FFmpeg extension, and avoid the OpenH264 decoder, if that's the source of the issue.

```
add-extensions:
  org.freedesktop.Platform.ffmpeg-full:
    directory: lib/ffmpeg
    version: '21.08'
    add-ld-path: .
    no-autodownload: false
...
modules:
- name: ...
   build-commands/post-install:
      - install -dm755 ${FLATPAK_DEST}/lib/ffmpeg

```

File: org.kde.kaffeine.json
Comment (hfiguiere) [inline org.kde.kaffeine.json:213]: usually name is the first property

**Comment by Mailaender** [comment]
bot, build org.kde.kaffeine

**Comment by travier** [comment]
Still does not work for me 😕

**Comment by tinywrkb** [comment]
For some reason, it doesn't work in Wayland, using `--socket=x11` removing the Wayland socket works. Maybe also unset `QT_QPA_PLATFORM` for users who have this set in the environment.

You might also need to patch VLC, as the plugins cache test fails. The following works for me, but you should look at what the VLC Flatpak has.

```
diff --git a/src/modules/bank.c b/src/modules/bank.c
index 2e67a0d..1e106e9 100644
--- a/src/modules/bank.c
+++ b/src/modules/bank.c
@@ -273,16 +273,6 @@ static int AllocatePluginFile (module_bank_t *bank, const char *abspath,
     if (bank->mode & CACHE_READ_FILE)
     {
         plugin = vlc_cache_lookup(&bank->cache, relpath);
-
-        if (plugin != NULL
-         && (plugin->mtime != (int64_t)st->st_mtime
-          || plugin->size != (uint64_t)st->st_size))
-        {
-            msg_Err(bank->obj, "stale plugins cache: modified %s",
-                    plugin->abspath);
-            vlc_plugin_destroy(plugin);
-            plugin = NULL;
-        }
     }
 
     if (plugin == NULL)
```

**Comment by Mailaender** [comment]
bot, build org.kde.kaffeine

**Comment by tinywrkb** [comment]
I looked at what the VLC Flatpak maintainer did. He kept the plugin size test (timestamp test still removed of course), but regenerated the plugins cache in the cleanup-commands step, which I'm guessing it's because cleanup runs after symbols extraction, so a plugin's size won't change anymore.  
I've no idea why the devs decided to cache timestamps and sizes, just version the plugins if you're worried about loading incorrect binaries, it's not like these values are tested if `VLC_PLUGIN_PATH` is set.

**Comment by Mailaender** [comment]
bot, build org.kde.kaffeine

**Comment by AsciiWolf** [comment]
bot, build org.kde.kaffeine

**Comment by AsciiWolf** [comment]
Tested and seems to work fine. :-)

The only problem that I noticed is desktop icon that is missing for some reason.

**Comment by travier** [comment]
I can play videos with the latest build! 🎉

**Comment by travier** [comment]
OK, the manifest needs a quick check/update to the latest releases and some x-checker-data config addition for dependencies but we should be good.

**Comment by AsciiWolf** [comment]
What about the missing desktop icon? :-)

**Comment by travier** [comment]
Sorry, I've not checked that.

**Comment by Mailaender** [comment]
I checked

> /app/share/icons/hicolor/scalable/apps/kaffeine.svg

is definitely installed

**Comment by AsciiWolf** [comment]
Then there is most likely something wrong with the AppData file. I took a quick look at the file and did not find any obvious issues. I think that there should not be a .desktop suffix in the [app id](https://github.com/KDE/kaffeine/blob/master/src/org.kde.kaffeine.appdata.xml#L3), but it should not do any harm (it is just not needed). Everything else looks good.

File: org.kde.kaffeine.json
Comment (AsciiWolf) [inline org.kde.kaffeine.json:15]:
```suggestion
  "command": "kaffeine",
  "rename-icon": "kaffeine",
```

This could help, but I am not sure.

Comment (hfiguiere) [reply org.kde.kaffeine.json:15]: I think it would.
Toward the end of the build I get:
```
WARNING: Icon not matching app id in /var/home/hub/source/flathub/org.kde.kaffeine/repo/export/share/applications/org.kde.kaffeine.desktop: kaffeine
WARNING: Icon referenced in desktop file but not exported: kaffeine
```

Comment (hfiguiere) [reply org.kde.kaffeine.json:15]: The installed destkop file has `Icon=kaffeine` which `rename-icon` will fix.

**Comment by hfiguiere** [comment]
> Then there is most likely something wrong with the AppData file. I took a quick look at the file and did not find any obvious issues. I think that there should not be a .desktop suffix in the [app id](https://github.com/KDE/kaffeine/blob/master/src/org.kde.kaffeine.appdata.xml#L3), but it should not do any harm (it is just not needed). Everything else looks good.

While it's better to not have it as it is deprecated we actually tolerate it.

**Comment by hfiguiere** [comment]
```
21-08-22 15:34:02.955 [Warning ] kf.solid.backends.udisks2: /run/build-runtime/solid/src/solid/devices/backends/udisks2/udisksmanager.cpp#160: void Solid::Backends::UDisks2::Manager::introspect(const QString&, bool): Failed enumerating UDisks2 objects: "org.freedesktop.DBus.Error.Disconnected" 
 "Not connected to D-Bus server"
```
it probably want udisk dbus

**Comment by Mailaender** [comment]
bot, build org.kde.kaffeine

**Comment by hfiguiere** [comment]
adding  `"--system-talk-name=org.freedesktop.UDisks2"`, fixes the error

**Comment by Mailaender** [comment]
I know. Figured it out yesterday in https://github.com/flathub/org.kde.kompare/pull/9.

**Comment by hfiguiere** [comment]
it also need     "--system-talk-name=org.freedesktop.UPower"

**Comment by Mailaender** [comment]
bot, build org.kde.kaffeine

**Comment by AsciiWolf** [comment]
I can confirm that there is now a correct desktop icon displayed. :-)

**Comment by travier** [comment]
The AppData patch lists `22.04.1` but the sources are at `2.0.18`. Commented there: https://invent.kde.org/multimedia/kaffeine/-/merge_requests/6 too.

**Comment by travier** [comment]
Should be good to go apart from that.

**Comment by Mailaender** [comment]
https://invent.kde.org/multimedia/kaffeine/-/merge_requests/7

**Comment by Mailaender** [comment]
bot, build org.kde.kaffeine

**Review by travier — APPROVED** [review]

**Comment by Mailaender** [comment]
Whoever had this brilliant idea that XML had to be manually sorted.

**Comment by tsdgeos** [comment]
22-08-22 21:24:20.340 [Warning ] kf.notifications: /run/build-runtime/knotifications/src/kstatusnotifieritem.cpp#998: void KStatusNotifierItemPrivate::setLegacySystemTrayEnabled(bool): env says KDE is running but SNI unavailable -- check KDE_FULL_SESSION and XDG_CURRENT_DESKTOP

Probably want --talk-name=org.kde.StatusNotifierWatcher ?

**Comment by Mailaender** [comment]
bot, build org.kde.kaffeine

**Comment by tsdgeos** [comment]
Have you checked if the digital tv stuff works? I wonder if there's special hardware access needed for that

**Comment by AsciiWolf** [comment]
> Have you checked if the digital tv stuff works? I wonder if there's special hardware access needed for that

I am also interested in this (but don't have any hardware that I could test with at the moment), but I am afraid that it will require custom udev rules on the host.

**Comment by tsdgeos** [comment]
I'm not 100% sure it makes a lot of sense to commit this if we can't verify the features work well, maybe merge to beta branch?

**Comment by AsciiWolf** [comment]
I don't personally consider the digital tv functionality as primary function of Kaffeine. Everything else seems to work fine. Also, every dvb-t stick that has its udev rules upstreamed in udevd (systemd) should theoretically work fine out-of-box.

**Comment by hfiguiere** [comment]
if we blocked merge any app that has stuff not working properly, then we'd have nothing

**Comment by travier** [comment]
I think we're in a state where the app is completely fine for beta.
I'd also argue that it has enough features working for stable.
Given that it's not even part of the release service, I'm not sure how much work will happen on this one so I'd be inclined to move it to stable and see if folks complain.

**Comment by tsdgeos** [comment]
> if we blocked merge any app that has stuff not working properly, then we'd have nothing

Sad though, giving people things that work worse than they could.

I won't block this going to regular release.

**Comment by travier** [comment]
Let's give it try. @nedrichards @barthalion Can you merge this one? Thanks!

**Pull request closed without merging**
